plugins/kotlin/formatter/src/org/jetbrains/kotlin/idea/formatter/KotlinCommonBlock.kt

```diff
@@ -1067,7 +1067,16 @@ private val INDENT_RULES = arrayOf(
 
     strategy("Block in when entry")
         .within(WHEN_ENTRY)
-        .notForType(BLOCK, WHEN_CONDITION_EXPRESSION, WHEN_CONDITION_IN_RANGE, WHEN_CONDITION_IS_PATTERN, ELSE_KEYWORD, ARROW)
+        .notForType(BLOCK, WHEN_CONDITION_EXPRESSION, WHEN_CONDITION_IN_RANGE, WHEN_CONDITION_IS_PATTERN, ELSE_KEYWORD)
+        .set(Indent.getNormalIndent()),
+
+    strategy("Block after wrapped arrow in when entry")
+        .within(WHEN_ENTRY)
+        .notForType(WHEN_CONDITION_EXPRESSION, WHEN_CONDITION_IN_RANGE, WHEN_CONDITION_IS_PATTERN, ELSE_KEYWORD, ARROW)
+        .forElement { node ->
+            val prevNode = node.prev() ?: return@forElement false
+            prevNode.elementType == ARROW && hasLineBreakBefore(prevNode)
+        }
         .set(Indent.getNormalIndent()),
```

Comment:
Here I've tried to add indentation after wrapped arrow:
```kotlin
fun foo(x: Any) = when (x) {
    Comparable::class, Iterable::class, String::class, // trailing comma
        -> {
            println(1)
        }
}
```
But I can't get it work. Code always looks like:
```kotlin
fun foo(x: Any) = when (x) {
    Comparable::class, Iterable::class, String::class, // trailing comma
        -> {
        println(1) // <- BLOCK has indent but it is not added to its children 
    }
}
```
Am I trying to do it right way?

Reply:
Fixed it with passing `enforceIndentToChildren = true`. Thanks to @Wamae fo this hint.

Reply:
I think this change is redundant, because 
```kotlin
fun foo(x: Any) = when (x) {
    Comparable::class, Iterable::class, String::class, // trailing comma
        -> {
        println(1)
    }
}
```
is expected code

Reply:
Removed this strategy